torch/__init__.py

```diff
@@ -2011,6 +2013,21 @@ def __getattr__(name):
 
         raise AttributeError(f"module '{__name__}' has no attribute '{name}'")
 
+def get_device_module(device):
```

Comment:
I think that for the general case we can follow the approach from https://github.com/pytorch/pytorch/blob/497bac223c72c14df85353512a77e60f3c8a6290/torch/_utils.py#L894
(we should also remove that private function and use this one in a follow up).

Also we should make sure that this function accepts any Device-like input and do appropriate typechecking.